src/apps/loyalty/server/index.tsx

```diff
@@ -72,53 +72,65 @@ app.get("/inquiries", (req, res) => {
     return res.redirect(req.baseUrl + "/login")
   }
 
-  let promise = IsomorphicRelay.prepareData({
-    Container: Inquiries,
-    queryConfig: new CurrentUserRoute(),
-  }, res.locals.networkLayer)
+  fetchCollectorProfile(req.user.get("accessToken"))
+    .then(body => body.json())
+    .then(info => {
+      if (info.loyalty_applicant_at) {
+        return res.redirect(req.baseUrl + "/thank-you")
+      }
 
-  promise
-    .then(({data, props}) => {
-      const html = renderToString(<IsomorphicRelay.Renderer {...props} />)
-      const styles = styleSheet.rules().map(rule => rule.cssText).join("\n")
-      res.send(renderPage({
-        styles,
-        html,
-        entrypoint: "/bundles/inquiries.js",
-        bootstrapData: `var DATA = ${JSON.stringify(data)}; var USER_DATA = ${JSON.stringify(req.user.toJSON())};`,
-      }))
+      if (info.confirmed_buyer_at) {
+        markCollectorAsLoyaltyApplicant(req.user.get("accessToken"))
+          .then(body => body.json())
+          .then(profile => {
+            return res.redirect(req.baseUrl + "/thank-you")
+          })
+          .catch(err => console.error(err))
+      }
+
+      let promise = IsomorphicRelay.prepareData({
+        Container: Inquiries,
+        queryConfig: new CurrentUserRoute(),
+      }, res.locals.networkLayer)
+
+      promise
+        .then(({data, props}) => {
+          const html = renderToString(<IsomorphicRelay.Renderer {...props} />)
+          const styles = styleSheet.rules().map(rule => rule.cssText).join("\n")
+          res.send(renderPage({
+            styles,
+            html,
+            entrypoint: "/bundles/inquiries.js",
+            bootstrapData: `var DATA = ${JSON.stringify(data)}; var USER_DATA = ${JSON.stringify(req.user.toJSON())};`,
+          }))
+        })
```

Comment:
warning, incoming untested psuedo-code. _If_ you want to look at using `async` / `await` for this code, it would look something like this:

```ts
app.get("/inquiries", async (req, res) => {
  if (!req.user) {
    return res.redirect(req.baseUrl + "/login")
  }

    const token = req.user.get("accessToken")
  const results = fetchCollectorProfile(token)
  const info = await results.json()

  if (info.loyalty_applicant_at) {
    return res.redirect(req.baseUrl + "/thank-you")
  }

  if (info.confirmed_buyer_at) {
      try {
        const applicant = markCollectorAsLoyaltyApplicant(token)
        const profile = applicant.json()
        return res.redirect(req.baseUrl + "/thank-you")
      } catch(err)
        console.error(err)
      }
  }

  const {data, props} = await IsomorphicRelay.prepareData({
    Container: Inquiries,
    queryConfig: new CurrentUserRoute(),
  }, res.locals.networkLayer)

  const html = renderToString(<IsomorphicRelay.Renderer {...props} />)
  const styles = styleSheet.rules().map(rule => rule.cssText).join("\n")
  
  res.send(renderPage({
    styles,
    html,
    entrypoint: "/bundles/inquiries.js",
    bootstrapData: `var DATA = ${JSON.stringify(data)}; var USER_DATA = ${JSON.stringify(req.user.toJSON())};`,
  }))
})
```